### PR TITLE
WT-13644 Implement new directory structure for modular testing

### PR DESCRIPTION
**Pull request by jiechenbo**
This ticket deals with creating a new directory structure to accommodate for modular testing. I have followed the structure followed in the ticket.

### CODE REVIEW

File: test/catch2/misc_tests/test_bounds_restore.cpp
Comment (luke-pearson) [inline test/catch2/misc_tests/test_bounds_restore.cpp:1]: Move this to cursor/unit.

File: test/catch2/README.md
Comment (luke-pearson) [inline test/catch2/README.md:23]: Reword: Module specific tests are found under the directory matching the modules name. That directory contains unit tests under unit/ and api contract tests under api/.

File: test/catch2/README.md
Comment (luke-pearson) [inline test/catch2/README.md:33]: The line width here can be fixed seeing as we're making big changes.

**Comment by luke-pearson** [comment]
This change is likely to break things unexpectedly so prior to merge please run the full WT test suite on it.

**Comment by jeremythorp** [comment]
Hi @jiechenbo , as my PR https://github.com/wiredtiger/wiredtiger/pull/11122 makes some fixes to the Catch2 tests, we should merge my PR first, before merging this PR. Thanks, Jeremy

**Comment by jeremythorp** [comment]
Hi @jiechenbo , even though this is a test-only change, please can you complete the "What makes this change safe?" section in the PR template as that's something we want completed for every PR. Thanks, Jeremy

File: test/catch2/README.md
Comment (jeremythorp) [inline test/catch2/README.md:1]: When using 'Catch2' as a word describing the unit test framework, the correct spelled is with a capital 'C' (see https://github.com/catchorg/Catch2)

Comment (jeremythorp) [reply test/catch2/README.md:1]: I also suggest adding a sentence here that says what Catch2 is and linking to https://github.com/catchorg/Catch2 .

File: test/catch2/README.md
Comment (jeremythorp) [inline test/catch2/README.md:24]: 'an unit' should be 'a unit'.

File: test/catch2/README.md
Comment (jeremythorp) [inline test/catch2/README.md:33]: 'want add' should be 'want to add'

File: test/catch2/README.md
Comment (jeremythorp) [inline test/catch2/README.md:38]: 'of' in 'all of the steps' can be removed.

**Comment by jeremythorp** [comment]
Hi @jiechenbo , 

I've proposed some small changes, and I don't have any other changes than those. 

Once Luke and my changes are in place, the conflicts resolved, and the changes from https://github.com/wiredtiger/wiredtiger/pull/11122 have been merged to develop and this branch, it should be good. I'll review again just as a double-check before approving.

Thanks, Jeremy

**Comment by jiechenbo** [comment]
I will wait for other tickets to be merged in first, fix the conflicts and running a full test suite before I re-request review

**Review by luke-pearson — APPROVED** [review]
LGTM. Please run a full patch before merge.

**Review by jeremythorp — APPROVED** [review]
LGTM!

**Pull request merged**
